lib/content_dumper.rb

```diff
@@ -0,0 +1,51 @@
+require "csv"
+require "zlib"
+
+class ContentDumper
+  FIELDS = %w(base_path content_id locale document_type schema_name rendering_app publishing_app updated_at).freeze
+  HASH_FIELDS = %w(details expanded_links).freeze
+
+  def initialize(filename)
+    @filename = filename
+  end
+
+  def dump
+    Zlib::GzipWriter.open(filename) do |file|
+      csv = CSV.new(file)
+      csv << csv_field_names
+      ContentItem.each do |row|
+        csv << csv_fields(row)
+      end
+    end
+  end
+
+private
+
+  attr_reader :filename
+
+  def csv_field_names
+    FIELDS + HASH_FIELDS.map { |field| "#{field}_hash".to_sym }
+  end
+
+  def csv_fields(item)
+    FIELDS.map { |field| item.send(field) } + HASH_FIELDS.map { |field| hash_field(item.send(field)) }
+  end
+
+  def hash_field(object)
+    Digest::MD5.hexdigest(
+      JSON.generate(
+        deep_sort_hash(object)
```

Comment:
Just to add, if you've seen problems with ordering that this deep sort fixes might want to ignore my suggestion. I'm working on assumption that this is pre-empting a problem rather than fixing one.